### ISSUE TITLE
Also alllow TemplateBased naming (for arguments etc.) placeholders in test description

### ISSUE THREAD

**Issue by rklec** [nunit/nunit#4595]
## Underlying Problem

I want to parametrize tests.

### Original problem: Description for tests

NCrunch seems to display the Description instead of the TestName if that is set. So such a test looks like this:

```csharp
        [TestCase(1, 1, Description = "ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit")]
        [TestCase(2, 2, Description = "ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit")]
        [TestCase(-1, -1, Description = "ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit")]
        [TestCase(1, 0)]
        public void
            ShouldBeTrue_WhenLastFoundTimeIsInThePast(int numberOfDaysLastFound, int notFoundLimitInDays)
        {
```

![grafik](https://github.com/nunit/nunit/assets/71322635/80f3d501-5240-4309-bd79-22ca6cf19ec1)

**The problem:** Now `ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit` is just duplicated, I cannot differentiate the tests and see them in NUnit.

### Try 1: Description for tests

There exist [**template-based placeholders**](https://docs.nunit.org/articles/nunit/running-tests/Template-Based-Test-Naming.html) like `{a}`:

```csharp
        [TestCase(1, 1, Description = "ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit{a}")]
        [TestCase(2, 2, Description = "ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit{a}")]
        [TestCase(-1, -1, Description = "ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit{a}")]
        [TestCase(1, 0)]
        public void
            ShouldBeTrue_WhenLastFoundTimeIsInThePast(int numberOfDaysLastFound, int notFoundLimitInDays)
        {
```
![grafik](https://github.com/nunit/nunit/assets/71322635/87862955-099c-4ccd-9fdf-6855488fbc20)

**The problem:** This does not work.

### Try 2: Name of tests (workaround)

You can change it to use `TestName` and then you can use [**template-based placeholders**](https://docs.nunit.org/articles/nunit/running-tests/Template-Based-Test-Naming.html) like `{a}`:

![grafik](https://github.com/nunit/nunit/assets/71322635/fba2982c-d450-494d-a1a0-187bbedcf53f)

```csharp
        [TestCase(1, 1, TestName = "ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit{a}")]
        [TestCase(2, 2, TestName = "ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit{a}")]
        [TestCase(-1, -1, TestName = "ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit{a}")]
        [TestCase(1, 0)]
        public void
            ShouldBeTrue_WhenLastFoundTimeIsInThePast(int numberOfDaysLastFound, int notFoundLimitInDays)
        {     
```

**The problem:** The linked NUnit doc strongly recommends _against_ this practice:
> **We strongly recommend you to NOT change the test naming.**

## Proposed solution

Basically make "Try 1" above work, i.e. also support placeholders like `{a}` in the test description.

**Comment by OsirisTerje** [comment]
The reason for the recommendation to not change the testname is that the Visual Studio test explorer is driven by the testname.  However, the default is `{m}{a}`, so not sure why that should not work for you.

> The problem: Now ShouldBeTrue_WhenIsLastFoundAtSameDayAsLimit is just duplicated, I cannot differentiate the tests and see them in NUnit.

I can't see what you're pointing at.  I am possibly drowning in the image there.  The first image looks like it still has template parameters, but the test code doesn't.  I am confused.  Can you point it out?

You are correct that the template parameters are for the testname, and not for the description.  I am not sure if you want this to work in VS or in NCrunch.

**Comment by rklec** [comment]
>  I am possibly drowning in the image there. [...]  I am confused. Can you point it out?

Oh I am sorry seems I indeed uploaded the wrong first image I am sorry, I've corrected it in the post.

So the actual problem seems to be that NCrunch _always shows the description.


Also to see how NCrunch structures tests the code structure in my example is:
```csharp
[TestOf(typeof(NotFoundWithinDuration))]
internal class NotFoundWithinDurationTest
{
    // ...
    public class ShouldDelete : NotFoundWithinDurationTest
    {
      // ...

      // ....
      public void
            ShouldBeTrue_WhenLastFoundTimeIsInThePast(int numberOfDaysLastFound, int notFoundLimitInDays)
        {
           // ....
        }
   }
```

So that is why there is a plus in there etc, it uses the class names for it. That said, this does not matter, actually. NCrunch just always replaces the test name with the NUnit description if it has one. That is the root cause, leading to my tries here...

NCrunch v4.11.0.2 (in my case) with Visual Studio 2022 (17.8.4)
